docs/pages/deploy-a-cluster/teleport-enterprise/gcp-kms.mdx

```diff
@@ -0,0 +1,270 @@
+---
+title: Google KMS
+description: Configure Teleport to store CA private keys in the Google Cloud Key Management Service
+h1: Store Teleport Private Keys in Google Cloud KMS
+---
+
+This guide will show you how to set up your Teleport Cluster to use the Google
+Cloud Key Management Service (KMS) to store and handle the CA private key
+material used to sign all certificates issued by your Teleport cluster.
+
+Teleport generates User, Host, JWT, and Database Certificate Authorities (CAs)
+during the first Auth Server's initial startup.
+These CAs are used to sign all certificates issued to clients and hosts in the
+Teleport cluster.
+When configured to use Google KMS, all private key material for these CAs will
+be generated, stored, and used for signing inside of Google KMS.
+In short, private key material will never leave Google KMS.
+Instead of the actual private key, Teleport will only store the ID of the KMS
+key.
+
+If launching a new Teleport cluster this will all be handled during initial
+startup with no specific interventions required after configuration. If
+configuring an existing Teleport cluster to switch to KMS keys after it has
```

Comment:
I would consider phrasing this as a security feature so it sounds less like an inconvenience. Maybe something like:

> For existing Teleport clusters that already have private key material on disk,
> a CA rotation must be performed. This ensures that the private key material
> only ever exists in the KMS when this feature is enabled.